base/linalg/triangular.jl

```diff
@@ -39,10 +39,15 @@ for t in (:LowerTriangular, :UnitLowerTriangular, :UpperTriangular, :UnitUpperTr
 
         real{T<:Real}(A::$t{T}) = A
         real{T<:Complex}(A::$t{T}) = (B = real(A.data); $t(B))
-
+        abs(A::$t) = $t(abs(A.data))
     end
 end
 
+imag(A::UpperTriangular) = UpperTriangular(imag(A.data))
+imag(A::LowerTriangular) = LowerTriangular(imag(A.data))
+imag(A::UnitLowerTriangular) = imag(tril(A,-1))
+imag(A::UnitUpperTriangular) = imag(triu(A,-1))
```

Comment:
`triu(A,+1)` here, right? This also makes two copies, I think `UpperTriangular(triu!(imag(A.data), 1))` would allocate less?